core/src/test/scala/unit/kafka/log/ProducerStateManagerTest.scala

```diff
@@ -900,6 +900,14 @@ class ProducerStateManagerTest {
     assertEquals(Seq(42), ProducerStateManager.listSnapshotFiles(logDir).map(_.offset).sorted)
   }
 
+  @Test
+  def testRemoveAndMarkSnapshotForDeletion(): Unit = {
```

Comment:
Could we add a test for the case where `snapshot.renameTo` fails due to the missing file?